File: Python/Lyrics.py
import requests
import string
from bs4 import BeautifulSoup

class Lyrics:
    def __init__(self, artist, title):
        self.title = title
        self.artist = artist
        self.lyrics = self.get_lyrics()

    # Web scrape from genius.com to get the lyrics of the song
    def get_lyrics(self):
        URL = self.get_url()
        r = requests.get(URL)
        soup = BeautifulSoup(r.content, 'html.parser')
        lyrics = soup.findAll('div', attrs = {'class':'lyrics'})
        self.lyrics = lyrics[0].text

    # Formats the url properly
    def get_url(self):
        temp_title = self.title.translate(str.maketrans(' ', '-', string.punctuation))
        temp_artist = self.artist.translate(str.maketrans(' ', '-', string.punctuation))
        url = "https://www.genius.com/%s-%s-lyrics" % (temp_artist, temp_title)
        return url

File: Java/Lyrics.java
import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;
import org.jsoup.nodes.TextNode;
import org.jsoup.nodes.Node;
import org.jsoup.Connection;
import java.io.IOException;

class Lyrics {

  public static String getUrl(String title, String artist) {
    String[] title_words = title.replaceAll("[^a-zA-Z0-9]"," ").toLowerCase().split("\\s+");
    String[] artist_words = artist.replaceAll("[^a-zA-Z0-9]"," ").toLowerCase().split("\\s+");
    title="";
    artist="";
    for (String word : title_words) {
      if (!word.equals(""))
        title+= word +"-";
    }
    for (String word : artist_words){
      if (!word.equals(""))
        artist+= word +"-";
    }
    String url = "https://www.genius.com/"+artist+title+"lyrics";
    return url;
  }

  public static String getLyrics(String url) {
    String lyrics = "";
    try {
      Document doc = Jsoup.connect(url).get();
      Element song = doc.select("div.lyrics").get(0);
      lyrics = song.wholeText();
      System.out.println(lyrics);
    }
    catch (IOException e) {
      System.out.println("Lyrics not found");
      return null;
    }
    return lyrics;
  }
}
